doc/rm/comportability_specification/index.md

```diff
@@ -249,9 +249,125 @@ See the section on [Inter Signal Handling](#inter-signal-handling) below for det
+
+Functionality is relevant to the TCB | Description of register functionality can be disclosed | Description of register functionality *cannot* be disclosed
+-------------------------------------|--------------------------------------------------------|-----------------------------------------------------------
+no                                   | Category A                                             | Category B
+yes                                  | Category A                                             | Category C
+
+- **Category A:** This category includes CSRs that may or may not be relevant to the TCB, but the mechanisms that they exercise are commonly known and their description can be disclosed.
+  As such, they can be implemented and documented like regular open-source registers.
+  Examples include driving strength or pull-up / pull-down enable registers for pad libraries, or design for test (DFT) registers steering a clock bypass mux.
+
+- **Category B:** This category includes CSRs that are not relevant to the TCB and whose description may not be disclosed.
+  Such registers must follow a specific register layout - however there are no restrictions in terms of programming model.
+  Examples include DFT and test related registers such as vendor-specific built-in self-test (BIST) structures.
+
+- **Category C:** This category includes CSRs relevant to the TCB whose description may not be disclosed.
+  Similarly to category B, such registers must follow a specific register layout and in addition the programming model needs to be structured such that these registers are compatible with an open-source TCB.
+  Examples include hardware calibration registers for mechanisms that cannot be described in open-source or chicken bits for security countermeasures.
+
+The definition and recommended programming model for a specific register pertaining to a closed-source IP depends on which of the above categories it belongs to, as described in the next two subsections.
+
+#### Register Node Structure
+
+In any case, the register node of a lowRISC peripheral IP containing the TL-UL bus interface and register address decoding logic must be located *outside* of the closed-source IP as illustrated below:
+
+![Types of Comportable IP](register_categories.svg)
+
+Such register nodes shall contain an open-source register region where regular open-source registers and registers belonging to category A can be implemented.
+
+For registers falling into categories B and C, a closed-source register region with a generic structure and description must be allocated.
+The structure may contain one or all the following generic sections:
+
+- A section of L x 32bit status registers with read-only SW access, labeled as STATUS,
+- A section of M x 32bit control registers with read-write SW access, labeled as CTRL,
+- A section of 1 x 1bit or M x 1bit write enable registers for the M control registers, labeled as CTRL_REGWEN,
```

Comment:
I have added the 1x1bit option here since it may not always make sense to define a multireg REGWEN.